compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2097,8 +2098,8 @@ impl<'a> Parser<'a> {
 
         let mut async_block_err = |e: &mut DiagnosticBuilder<'_>, span: Span| {
             recover_async = true;
-            e.span_label(span, "`async` blocks are only allowed in the 2018 edition");
-            e.help("set `edition = \"2018\"` in `Cargo.toml`");
+            e.span_label(span, "`async` blocks are only allowed in edition 2018 or later");
```

Comment:
For consistency with other errors, do you want to change this to one of the below? "are only allowed in edition 2018 or later" sounds a bit off to me, and it's inconsistent with some of the other errors.

```suggestion
            e.span_label(span, "`async` blocks are only allowed in the 2018 edition or later");
```

```suggestion
            e.span_label(span, "`async` blocks are only allowed in Rust 2018 or later");
```

Personally I think I prefer the second option.

(you'll need to bless tests as well)

Reply:
Sounds good. I'll update it, also for the `edition 2015` message.